mlir/lib/Dialect/Linalg/Transforms/ElementwiseOpFusion.cpp

```diff
@@ -927,17 +927,43 @@ fuseWithReshapeByExpansion(LinalgOp linalgOp, Operation *reshapeOp,
       iteratorTypes[j] = type;
 
   TypeRange resultTypes = ValueRange(outputs).getTypes();
-  auto fusedOp =
-      rewriter.create<GenericOp>(linalgOp.getLoc(), resultTypes,
-                                 /*inputs=*/expandedOpOperands, outputs,
-                                 expandedOpIndexingMaps, iteratorTypes);
-  Region &fusedRegion = fusedOp->getRegion(0);
-  Region &originalRegion = linalgOp->getRegion(0);
-  rewriter.cloneRegionBefore(originalRegion, fusedRegion, fusedRegion.begin());
-
-  // Update the index accesses after the expansion.
-  updateExpandedGenericOpRegion(rewriter, loc, fusedRegion, expansionInfo);
-
+  Operation *fusedOp;
+
+  TypeSwitch<Operation *>(linalgOp.getOperation())
```

Comment:
Could we move this into its own method.

Also the way I see it, we want to use clone for some operations (like `linalg.fill`/`linalg.copy`), the method you added for `linalg.transpose` and what is currently being done for `linalg.generic` as the default.

Reply:
yes, does make sense to default to generic and be more prescriptive when we want to retain the named op, fixed.